File: 二进制/693.cpp
class Solution {
public:
    string tenToTwo(int num) {
        string ans = "";

        while (num > 0) {
            ans = to_string(num % 2) + ans;
            num /= 2;
        }

        return ans;
    }

    bool hasAlternatingBits(int n) {
        string t = tenToTwo(n);

        bool flag = true;
        for (int i=1;i<t.size();i++) {
            if (t[i-1]-'0' == t[i]-'0') {
                flag = false;
                break;
            }
        }

        return flag;
    }
};

File: 力扣/二进制/693.java
class Solution {
    public String tenToTwo(int num) {
        StringBuilder sb = new StringBuilder();

        while (num > 0) {
            sb.append(num % 2);
            num /= 2;
        }
 
        return sb.toString();
    }

    public boolean hasAlternatingBits(int n) {
        String t = tenToTwo(n);

        boolean flag = true;
        for (int i=1;i<t.length();i++) {
            if (t.charAt(i-1)-'0' == t.charAt(i)-'0') {
                flag = false;
                break;
            }
        }

        return flag;
    }
}
